File: src/app/chop.ts
export class chop {
    id: number;
    name:String;
    prix:number;
    type:String;
    starRating:number;
    persentage:number;
    addedDate:Date;
    color:String;
    size:String;
    kontiter:number;
    imageslid1:string;
    imageslid2:string;
    imageslid3:string;
    imageslid4:string;
    constructor(data: any = {}) {
        this.id= data.id|| 0;
        this.name = data.name || '';
        this.prix = data.prix || 0;
        this.kontiter = data.kontiter || 0;
        this.type = data.type || '';
        this.color = data.color || '';
        this.size = data.size || '';
        this.persentage = data.persentage || 0;
        this.starRating = data.starRating || 0;
        this.addedDate = data.date;
        this.imageslid1 = data.imageslid1 || '';
        this.imageslid2 = data.imageslid2 || '';
        this.imageslid3 = data.imageslid3 || '';
        this.imageslid4 = data.imageslid4 || '';

      }
}

File: Back-end/src/main/java/com/example/demo/entity/Chop.java
package com.example.demo.entity;

import java.sql.Timestamp;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.Basic;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.Lob;
import javax.persistence.ManyToMany;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;


@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Entity
@Table(name = "chop")
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Chop {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY )
     private long id;
	 private String name;
	 private long prix;
	 private String type;
     private Timestamp addedDate;
     private long persentage;
     private long starRating;
     private long kontiter;
     private String size;
     private String color;
	@Lob
	@Basic(fetch = FetchType.LAZY)
	private byte[] imageslid1; 
	@Lob
	@Basic(fetch = FetchType.LAZY)
	private byte[] imageslid2; 
	@Lob
	@Basic(fetch = FetchType.LAZY)
	private byte[] imageslid3; 
	@Lob
	@Basic(fetch = FetchType.LAZY)
	private byte[] imageslid4; 
	
	@ManyToMany(fetch = FetchType.LAZY)
	@JoinTable(	name = "userchop", 
				joinColumns = @JoinColumn(name = "chop_id"), 
				inverseJoinColumns = @JoinColumn (name = "user_id"))
    private Set<User> userchop = new HashSet<>();

	@ManyToMany(fetch = FetchType.LAZY)
	@JoinTable(	name = "userpaymentchop", 
				joinColumns = @JoinColumn(name = "chop_id"), 
				inverseJoinColumns = @JoinColumn (name = "user_id"))
    private Set<User> userpaymentchop = new HashSet<>();
	
	
}
